lib/Sema/CSDiagnostics.cpp

```diff
@@ -6015,3 +6013,50 @@ bool UnableToInferClosureReturnType::diagnoseAsError() {
 
   return true;
 }
+
+static std::tuple<StringRef, StringRef>
+getImportModuleAndDefaultType(const ASTContext &ctx, ProtocolDecl *protocol) {
+  const auto &target = ctx.LangOpts.Target;
+  if (protocol ==
+      ctx.getProtocol(KnownProtocolKind::ExpressibleByColorLiteral)) {
+    if (target.isMacOSX()) {
+      return std::make_tuple("AppKit", "NSColor");
+    } else if (target.isiOS() || target.isTvOS()) {
+      return std::make_tuple("UIKit", "UIColor");
+    }
+  } else if (protocol ==
+             ctx.getProtocol(KnownProtocolKind::ExpressibleByImageLiteral)) {
+    if (target.isMacOSX()) {
+      return std::make_tuple("AppKit", "NSImage");
+    } else if (target.isiOS() || target.isTvOS()) {
+      return std::make_tuple("UIKit", "UIImage");
+    }
+  } else if (protocol ==
+             ctx.getProtocol(
+                 KnownProtocolKind::ExpressibleByFileReferenceLiteral)) {
+    return std::make_tuple("Foundation", "URL");
+  }
+  return std::make_tuple("", "");
```

Comment:
Is there any code path where this would be returned? Looking at the diagnostic changes it seems like we could make this an unreachable but I’m not completely sure.

Reply:
I'm not sure either, but since there was a code path in the old logic checking for empty to not emit the note, I thought the safer path was to just return empty.

Reply:
Hmm, given there are only three kinds of `ObjectLiteralExpr`, this should be unreachable.

Reply:
Also, it should be possible to check for it using `getLiteralKind` instead of checking the protocol.